### PR TITLE
fix(webhook): treat 404 and other error replies as errors in async requests

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-10405

The problem here was that, for async requests, ehttpc responses of the form `{ok, 4__, _, _}` and similar were being treated as successes.

### <samp>🤖 Generated by Copilot at 86ae4e0</samp>

This pull request enhances the testing and reliability of the web hook bridge and the HTTP connector modules. It adds a new dependency on `snabbkaffe`, a test framework for tracing and mocking, and uses it to test the error handling and retry logic of the web hook bridge. It also simplifies and logs the HTTP requests and responses in the HTTP connector, and adds trace points for testing. It fixes a syntax error in the bridge configuration file.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [na] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [na] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [na] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [na] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_connector/src/emqx_connector_http.erl [hunk 1]
```diff
@@ -657,14 +636,29 @@ transform_result(Result) ->
...
-        _ ->
-            Result
+        {error, _Reason} ->
+            Result;
+        {ok, StatusCode, _} when StatusCode >= 200 andalso StatusCode < 300 ->
+            Result;
+        {ok, StatusCode, _, _} when StatusCode >= 200 andalso StatusCode < 300 ->
+            Result;
+        {ok, StatusCode, Headers} ->
+            {error, {unrecoverable_error, #{status_code => StatusCode, headers => Headers}}};
```
Comment (keynslug) [inline apps/emqx_connector/src/emqx_connector_http.erl:646]: Just a thought: wonder if we should treat `429 Too Many Requests` as unrecoverable.

**Review by keynslug — APPROVED** [review]

File: apps/emqx_connector/src/emqx_connector_http.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_connector/src/emqx_connector_http.erl:646]: Good point, I think that would be an exception here.  I'll add a special clause for it.  Thanks!

Comment (thalesmg) [reply apps/emqx_connector/src/emqx_connector_http.erl:648]: :heavy_check_mark:

**Review by keynslug — APPROVED** [review]
🔥

**Review by ieQu1 — APPROVED** [review]

**Pull request merged**